### DISCUSSION TITLE
Is it possible to animate glyphs/ characters?

### DISCUSSION THREAD

**Discussion by mainrs** [elkowar/eww#385] [Q&A] [answered]
For example for a workspace widget. Each time the focused workspace changes, the unfocused icon fades out and the new one fades in.

**Comment by druskus20** [comment, marked as answer]
You could do this with some quite complicated revealer setup. 
I recommend defining a generic workspace widget. You'd have 9 (one per workspace) of those. 
Each one contains a revealer that reveals the contents if that particular workspace is focused.
Then you would have a workspaces script that outputs json with the information regarding each workspace

```clojure
; This might not be correct code, I'm writing this from memory, haven't touched eww in a while :) 
;   waiting for that localstate pr @elkowar :D :D 
(defwidget worspaces [] 
  (workspace :icon "1" :active workspaces_script.first.active)
  (workspace :icon "2" :active workspaces_script.second.active)
   ...
   ...)

(defwideget workspace [icon, active] 
... revealer ... )
```

Might do this myself actually at some point. I believe it's possible. Do take a look at https://github.com/druskus20/eugh for examples of revealers

**Reply by mainrs** [reply]
Fair enough! That should work for my use-case. Maybe this is a feature that _could_ be explored more down the line :)
